### ISSUE TITLE
Upgrade to latest did-jwt

### ISSUE THREAD

**Issue by theblockstalk** [Tonomy-Foundation/Tonomy-ID-SDK#336]
Upgrade to the latest version of this library Would provide additional security and functional upgrades, as well as put us in a position to start using the Veramo library. See task https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/issues/337

Why have we not done this yet:
-  there is a build issue when doing this upgrade. see https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/pull/234 and https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/issues/316
- basically, the unit8arrays and multiformats libraries are not exporting a compatible Package that can be bundled with the current jest bundler

Repositories
- [ ] did-antelope-resolver
- [ ] Tonomy SDK

Also need to upgrade:
- did-resolver
- did-jwt
- did-jwt-vc

**Comment by theblockstalk** [comment]
I have found 1 solution so far which allows resolution with jest, using the following in the jest.config.ts file which you can see in in PR commit https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/pull/394/commits/22e19041e002344fe31410962208dc896ed3dcbd

```ts
moduleNameMapper: {
        // This is a workaround for the following packages that use conditional exports in package.json which jest does not support yet
        "^uint8arrays$": "<rootDir>/node_modules/uint8arrays/esm/src/index.js",
        "^@ipld/dag-pb*": "<rootDir>/node_modules/@ipld/dag-pb/src/index.js",
        "^multiformats/(.*)$": "<rootDir>/node_modules/multiformats/dist/src/$1.js",
        "^ipfs-unixfs$": "<rootDir>/node_modules/ipfs-unixfs/dist/src/index.js",
        "^protons-runtime$": "<rootDir>/node_modules/protons-runtime/dist/src/index.js",
        "^uint8-varint$": "<rootDir>/node_modules/uint8-varint/dist/src/index.js",
        "^uint8arrays/(.*)$": "<rootDir>/node_modules/uint8arrays/esm/src/$1.js",
    }
```

**Comment by theblockstalk** [comment]
also tried, unsucessfully:
- upgrading all other packages, node version and yarn

changing jest.config.ts:
- adding "resolver": "jest-resolver-enhanced" to jest.config.js
- adding "resolver": "jest-resolve" to jest.config.js
- setting "preset": 'ts-jest/presets/default-esm',
- building a custom jest resolver. see below two different configurations tried:


jest-resolver.cjs
```js
const defaultResolver = require('jest-resolve').default;
const path = require('path');

// use the import field in package.json to resolve the module manually
module.exports = (request, options) => {
    // Attempt to resolve using the default resolver
    try {
        return defaultResolver(request, options);
    } catch (e) {
        // If the resolution fails, attempt to resolve using "exports"
        const modulePath = path.join(options.basedir, 'node_modules', request);
        const pkgPath = path.join(modulePath, 'package.json');

        try {
            const pkg = require(pkgPath);
            if (pkg.exports && pkg.exports.import) {
                return path.join(modulePath, pkg.exports.import);
            }
        } catch (innerError) {
            // If anything goes wrong, throw the original error
            throw innerError;
        }
    }
};

// explicitly specify the conditions to resolve the module
module.exports = (path, options) => {
    // Call the defaultResolver, so we leverage its cache, error handling, etc.
    return options.defaultResolver(path, {
        ...options,
        conditions: ["default", "import"]
    });
};
```

**Comment by theblockstalk** [comment]
See https://github.com/decentralized-identity/veramo/issues/1250#issuecomment-1821335010

**Comment by theblockstalk** [comment]
Good chat here with Veramo author
https://discord.com/channels/878293684620234752/1200026960307437578/1249720913390469245
